### PR TITLE
Prune sbom and attestation images from the registry

### PR DESCRIPTION
**Pull request by mkosiarc**
If application image is removed from quay, we need to make sure that the associated sbom and attestation images are removed as well.

This is done with cronjob that will run python script that checks registry repositories via API.

[STONEBLD-1251](https://issues.redhat.com//browse/STONEBLD-1251)

### CODE REVIEW

File: config/registry_image_pruner/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  name: image-pruner-cronjob
+spec:
+  schedule: "0 0 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: image-pruner
+            image: registry.redhat.io/rhel8/python-39
```
Comment (Michkov) [inline config/registry_image_pruner/cronjob.yaml:13]: bind to tag and sha

File: config/registry_image_pruner/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+apiVersion: batch/v1
+kind: CronJob
```
Comment (Michkov) [inline config/registry_image_pruner/cronjob.yaml:2]: maybe we want to disable concurency

File: config/registry_image_pruner/cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                  secretKeyRef:
+                    name: quaytoken
+                    key: organization
+            imagePullPolicy: IfNotPresent
+            command:
+              - /bin/bash
+              - '-c'
+              - >-
+                pip install -r /image-pruner/requirements.txt && python
+                /image-pruner/prune_images.py --namespace $(NAMESPACE) --dry-run
```
Comment (Michkov) [inline config/registry_image_pruner/cronjob.yaml:31]: dry run not needed

File: config/registry_image_pruner/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+namespace: image-controller-system
+
```
Comment (Michkov) [inline config/registry_image_pruner/kustomization.yaml:2]: include this file in config/base

Comment (mkosiarc) [reply config/registry_image_pruner/kustomization.yaml:2]: What do you mean? I don't see config/base

File: config/registry_image_pruner/cronjob.yaml (hunk 2 above)
Comment (mkosiarc) [reply config/registry_image_pruner/cronjob.yaml:2]: updated

File: config/registry_image_pruner/cronjob.yaml (hunk 3 above)
Comment (mkosiarc) [reply config/registry_image_pruner/cronjob.yaml]: was still leaving that for testing purposes, but updated.

File: config/registry_image_pruner/cronjob.yaml (hunk 1 above)
Comment (mkosiarc) [reply config/registry_image_pruner/cronjob.yaml]: binded to the newest version tag available

**Review by mmorhun — APPROVED** [review]
LGTM
Please add missing new lines at the end of files

File: config/registry_image_pruner/kustomization.yaml (hunk 4 above)
Comment (mmorhun) [reply config/registry_image_pruner/kustomization.yaml:2]: I think that `image-controller-system` should be "kustomizable".
In real clusters, usually, the namespace is `image-controller`.

Comment (mkosiarc) [reply config/registry_image_pruner/kustomization.yaml:2]: I am still lost :D

Comment (Michkov) [reply config/registry_image_pruner/kustomization.yaml:2]: Ok I meant https://github.com/redhat-appstudio/image-controller/blob/main/config/default/kustomization.yaml

Comment (Michkov) [reply config/registry_image_pruner/kustomization.yaml:2]: add `../registry_image_pruner` into bases and test by running `kustomize build config/default` that should generate the CRs also with the cronjob.

Comment (mkosiarc) [reply config/registry_image_pruner/kustomization.yaml]: I did that and also removed namespace and nameprefix from config/registry_image_pruner/kustomization.yaml as they are mentioned in the default and other kustomization files don't have them.

**Review by Michkov — APPROVED** [review]

**Pull request merged**